### PR TITLE
Fixes #2545 - Emit a click fact when selecting a context menu item.

### LINKED ISSUE #2545: Allow hooks for telemetry
In Fenix we want to have metrics for how users share content. https://github.com/mozilla-mobile/fenix/issues/957. I would love to have a way to know when an candidate action is invoked here: https://github.com/mozilla-mobile/android-components/blob/master/components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/ContextMenuFeature.kt#L113

I was thinking about adding a contextMenuItemWasSelected delegate method that I could hook into. What do you think?

### PR DESCRIPTION
**Pull request by boek**
I originally wanted to refactor this a little bit to be a little less stringly-typed. But that quickly grew out of scope of this PR. 

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/facts/ContextMenuFacts.kt [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    ).collect()
+}
+
+private object ContextMenuItems {
+    const val ITEM = "item"
+}
+
+internal fun emitClickFact(candidate: ContextMenuCandidate) {
+    val metadata = mapOf(
+            "item" to candidate.id
```
Comment (jonalmeida) [inline components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/facts/ContextMenuFacts.kt]: The spacing probably going to fail the linter. :)

Comment (boek) [reply components/feature/contextmenu/src/main/java/mozilla/components/feature/contextmenu/facts/ContextMenuFacts.kt]: @jonalmeida weird, it didn't! I will fix it anyways!

**Comment by boek** [comment]
Hmm.. Also looks like my editor removed all the trailing whitespace in the changelog. I can remove those changes if you'd like

**Comment by jonalmeida** [comment]
It's fine. I wish we had a linter for markdown to avoid the trailing spaces. :)

**Comment by jonalmeida** [comment]
Build failed on unrelated tasks. Restarting to merge..

**Comment by jonalmeida** [comment]
This is silly.. someone else needs to force merge it, sorry.

**Pull request merged**